ui/app/components/keymgmt/distribute.js

```diff
@@ -0,0 +1,233 @@
+  @tracked protection;
+}
+
+const VALID_TYPES_BY_PROVIDER = {
+  gcpckms: ['aes256-gcm96', 'rsa-2048', 'rsa-3072', 'rsa-4096', 'ecdsa-p256', 'ecdsa-p384', 'ecdsa-p521'],
+  awskms: ['aes256-gcm96'],
+  azurekeyvault: ['rsa-2048', 'rsa-3072', 'rsa-4096'],
+};
+export default class KeymgmtDistribute extends Component {
+  @service store;
+  @service flashMessages;
+  @service router;
+
+  @tracked keyModel;
+  @tracked isNewKey = false;
+  @tracked providerType;
+  @tracked formData;
+
+  constructor() {
+    super(...arguments);
+    this.formData = new DistributionData();
+    // Set initial values passed in
+    this.formData.key = this.args.key || '';
+    this.formData.provider = this.args.provider || '';
+    // Side effects to get types of key or provider passed in
+    if (this.args.provider) {
+      this.getProviderType(this.args.provider);
+    }
+    if (this.args.key) {
+      this.getKeyInfo(this.args.key);
+    }
+    this.formData.operations = [];
+  }
+
+  get keyTypes() {
+    return KEY_TYPES;
+  }
+
+  get validMatchError() {
+    if (!this.providerType || !this.keyModel?.type) {
+      return null;
+    }
+    const valid = VALID_TYPES_BY_PROVIDER[this.providerType]?.includes(this.keyModel.type);
+    if (valid) return null;
+
+    // default to showing error on provider unless @provider (field hidden)
+    if (this.args.provider) {
+      return {
+        key: `This key type is incompatible with the ${this.providerType} provider. To distribute to this provider, change the key type or choose another key.`,
+      };
+    }
+
+    const message = `This provider is incompatible with the ${this.keyModel.type} key type. Please choose another provider`;
+    return {
+      provider: this.args.key ? `${message}.` : `${message} or change the key type.`,
+    };
+  }
+
+  get operations() {
+    const pt = this.providerType;
+    if (pt === 'awskms') {
+      return ['encrypt', 'decrypt'];
+    } else if (pt === 'gcpckms') {
+      const kt = this.keyModel?.type || '';
+      switch (kt) {
+        case 'aes256-gcm96':
+          return ['encrypt', 'decrypt'];
+        case 'rsa-2048':
+        case 'rsa-3072':
+        case 'rsa-4096':
+          return ['decrypt', 'sign'];
+        case 'ecdsa-p256':
+        case 'ecdsa-p384':
+          return ['sign'];
+        default:
+          return ['encrypt', 'decrypt', 'sign', 'verify', 'wrap', 'unwrap'];
+      }
+    }
+
+    return ['encrypt', 'decrypt', 'sign', 'verify', 'wrap', 'unwrap'];
+  }
+
+  async getKeyInfo(keyName, isNew = false) {
+    let key;
+    if (isNew) {
+      this.isNewKey = true;
+      key = await this.store.createRecord(`keymgmt/key`, {
+        backend: this.args.backend,
+        id: keyName,
+        name: keyName,
+      });
+    } else {
+      key = await this.store
+        .queryRecord(`keymgmt/key`, {
+          backend: this.args.backend,
+          id: keyName,
+          recordOnly: true,
+        })
+        .catch(() => {});
```

Comment:
My thought here is that if the key isn't able to be read for some reason, since it's a non-essential side effect we should not break the interface and let them attempt to continue distributing the key. If the key/provider type is mismatched we'll let the API respond with that information rather than providing it here since they don't have permissions. This is a good reminder to capture some of this in the swallowed catch block